go.mod

```diff
@@ -60,8 +60,6 @@ require (
 
 replace mellium.im/sasl => github.com/mellium/sasl v0.2.1
 
-replace github.com/onflow/flow-go => ./
-
+// needed otherwise the code refers to the crypto package will refer to
+// the released version of the crypto library, which might not be the latest
 replace github.com/onflow/flow-go/crypto => ./crypto
```

Comment:
Instead of this replace statement, crypto should be bumped to `crypto/v0.11.0`

Reply:
I agree with Bastian on this. We should pin a specific version and keep it up to date. Otherwise importers of `flow-go` could have crypto-dependent `flow-go` packages that either don't compile or are invisibly using an older version of `crypto`.

Reply:
I agree. I was trying that, but for some reason, it didn't work.

The version for `crypto` has been [pinned](https://github.com/onflow/flow-go/pull/137/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R36) here. Suppose we no longer need this `replace`. But if I remove it, `make lint` will start complaining. 

Not sure what's the proper fix.

Reply:
I guess this is why: https://github.com/onflow/flow-go/blob/f84b80f81e683f18ba802015b0167ee6618a4f62/crypto/README.md#L16-L26

When we pin a version, Go will download a new copy in `pkg/mod` and that copy hasn't had the extra C build steps applied to it. We would need to follow @tarakby's instructions for the copy managed by `go mod`. Probably we would need to do that every time we change the version. From a bit of searching it doesn't really seem like Go has a good solution for modules that require a non-Go build step.

Reply:
I think there is another reason for @zhangchiqing's linter issues that need to be solved.
When we `go get` Flow crypto lib, the C source code of the package is copied along with Go files to `pkg/mod`. What I recommended in the readme is only related to the Relic dependency.